google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1beta2/CivilTimeEncoder.java

```diff
@@ -0,0 +1,338 @@
+  private static LocalTime decodePacked32TimeSeconds(int bitFieldTimeSeconds) {
+    checkValidBitField(bitFieldTimeSeconds, TIME_SECONDS_MASK);
+    int hourOfDay = getFieldFromBitField(bitFieldTimeSeconds, HOUR_MASK, HOUR_SHIFT);
+    int minuteOfHour = getFieldFromBitField(bitFieldTimeSeconds, MINUTE_MASK, MINUTE_SHIFT);
+    int secondOfMinute = getFieldFromBitField(bitFieldTimeSeconds, SECOND_MASK, SECOND_SHIFT);
+    // LocalTime validates the input parameters.
+    try {
+      return LocalTime.of(hourOfDay, minuteOfHour, secondOfMinute);
+    } catch (DateTimeException e) {
+      throw new IllegalArgumentException(e.getMessage(), e);
+    }
+  }
+
+  /**
+   * Encodes {@code time} as a 8-byte integer with microseconds precision.
+   *
+   * <p>Encoding is as the following:
+   *
+   * <pre>
+   *        6         5         4         3         2         1
+   * MSB 3210987654321098765432109876543210987654321098765432109876543210 LSB
+   *                                | H ||  M ||  S ||-------micros-----|
+   * </pre>
+   *
+   * @see #decodePacked64TimeMicros(long)
+   * @see #encodePacked64TimeMicros(LocalTime)
+   */
+  @SuppressWarnings("GoodTime")
+  public static long encodePacked64TimeMicros(LocalTime time) {
+    checkValidTimeMillis(time);
+    return (((long) encodePacked32TimeSeconds(time)) << MICRO_LENGTH) | (time.getNano() / 1_000L);
+  }
+
+  /**
+   * Decodes {@code bitFieldTimeMicros} as a {@link LocalTime} with microseconds precision.
+   *
+   * <p>Encoding is as the following:
+   *
+   * <pre>
+   *        6         5         4         3         2         1
+   * MSB 3210987654321098765432109876543210987654321098765432109876543210 LSB
+   *                                | H ||  M ||  S ||-------micros-----|
+   * </pre>
+   *
+   * @see #encodePacked64TimeMicros(LocalTime)
+   */
+  @SuppressWarnings("GoodTime-ApiWithNumericTimeUnit")
+  public static LocalTime decodePacked64TimeMicros(long bitFieldTimeMicros) {
+    checkValidBitField(bitFieldTimeMicros, TIME_MICROS_MASK);
+    int bitFieldTimeSeconds = (int) (bitFieldTimeMicros >> MICRO_LENGTH);
+    LocalTime timeSeconds = decodePacked32TimeSeconds(bitFieldTimeSeconds);
+    int microOfSecond = getFieldFromBitField(bitFieldTimeMicros, MICRO_MASK, MICRO_SHIFT);
+    checkValidMicroOfSecond(microOfSecond);
+    return timeSeconds.withNano(microOfSecond * 1_000);
+  }
+
+  /**
+   * Encodes {@code dateTime} as a 8-byte integer with seconds precision.
+   *
+   * <p>Encoding is as the following:
+   *
+   * <pre>
+   *        6         5         4         3         2         1
+   * MSB 3210987654321098765432109876543210987654321098765432109876543210 LSB
+   *                             |--- year ---||m || D || H ||  M ||  S |
+   * </pre>
+   *
+   * @see #decodePacked64DatetimeSeconds(long)
+   */
+  @SuppressWarnings("GoodTime-ApiWithNumericTimeUnit")
+  public static long encodePacked64DatetimeSeconds(LocalDateTime dateTime) {
```

Comment:
Do we need this to be public?

Reply:
But for the other case, we put it as a private method?

Reply:
I just realized you're totally right, we basically just need encode/decode for datetime micros and time micros don't we. The user won't actually need to use the seconds methods. I'll make both encode and decode for seconds private